File: backend/src/main/java/dev/anthonybruno/gymbuddy/user/UserServiceImpl.java
package dev.anthonybruno.gymbuddy.user;

import dev.anthonybruno.gymbuddy.user.model.UserDetails;

public class UserServiceImpl implements UserService {

    UserRepository userRepository = new UserRepository();

    @Override
    public UserDetails addUser(String username, String password, String email) {
        userRepository.addUser(username, password, email);
        return new UserDetails(0, username, email);
    }

    @Override
    public UserDetails editUser(long userId, UserDetails newDetails) {
        return null;
    }

    @Override
    public UserDetails getUser(long userId) {
        return null;
    }

    @Override
    public void deleteUser(long userId) {

    }
}

File: backend/src/main/kotlin/dev/anthonybruno/gymbuddy/user/UserServiceImpl.kt
package dev.anthonybruno.gymbuddy.user;

import dev.anthonybruno.gymbuddy.user.model.UserDetails;

public class UserServiceImpl implements UserService {

    UserRepository userRepository = new UserRepository();

    @Override
    public UserDetails addUser(String email, String password) {
        userRepository.addUser(email, password);
        return new UserDetails(0, email);
    }

    @Override
    public UserDetails editUser(long userId, UserDetails newDetails) {
        throw new UnsupportedOperationException();
    }

    @Override
    public UserDetails getUser(long userId) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void deleteUser(long userId) {
        throw new UnsupportedOperationException();
    }
}
